src/Native/System.Native/pal_networking.cpp

```diff
@@ -13,6 +13,19 @@
 #include <sys/socket.h>
 #include <unistd.h>
 #include <vector>
+#include <errno.h>
+
+#if defined(__APPLE__)
```

Comment:
Yup, without guard, FreeBSD with Clang37 complains:

```
/root/corefx/src/Native/System.Native/pal_networking.cpp:731:35: error: use of undeclared identifier 'IPV6_ADD_MEMBERSH$
            optionName = isIPv6 ? IPV6_ADD_MEMBERSHIP : IP_ADD_MEMBERSHIP;
                                  ^
```

There are other errors besides this, but I think they are out of this PR's scope. :)